spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/utils/GravitinoTableInfoHelper.java

```diff
@@ -3,73 +3,54 @@
 import com.datastrato.gravitino.spark.connector.ConnectorConstants;
 import com.datastrato.gravitino.spark.connector.PropertiesConverter;
 import com.datastrato.gravitino.spark.connector.SparkTransformConverter;
 import com.datastrato.gravitino.spark.connector.SparkTypeConverter;
-import com.google.common.annotations.VisibleForTesting;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
-import java.util.Set;
 import java.util.stream.Collectors;
-import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
 import org.apache.spark.sql.connector.catalog.Identifier;
-import org.apache.spark.sql.connector.catalog.SupportsRead;
-import org.apache.spark.sql.connector.catalog.SupportsWrite;
-import org.apache.spark.sql.connector.catalog.Table;
-import org.apache.spark.sql.connector.catalog.TableCapability;
-import org.apache.spark.sql.connector.catalog.TableCatalog;
 import org.apache.spark.sql.connector.expressions.Transform;
-import org.apache.spark.sql.connector.read.ScanBuilder;
-import org.apache.spark.sql.connector.write.LogicalWriteInfo;
-import org.apache.spark.sql.connector.write.WriteBuilder;
 import org.apache.spark.sql.types.DataTypes;
 import org.apache.spark.sql.types.Metadata;
 import org.apache.spark.sql.types.MetadataBuilder;
 import org.apache.spark.sql.types.StructField;
 import org.apache.spark.sql.types.StructType;
-import org.apache.spark.sql.util.CaseInsensitiveStringMap;
 
 /**
- * Provides schema info from Gravitino, IO from the internal spark table. The specific catalog table
- * could implement more capabilities like SupportsPartitionManagement for Hive table, SupportsIndex
- * for JDBC table, SupportsRowLevelOperations for Iceberg table.
+ * GravitinoTableInfoHelper is a common helper class that is used to retrieve table info from the
+ * Gravitino Server
  */
-public abstract class SparkBaseTable implements Table, SupportsRead, SupportsWrite {
+public class GravitinoTableInfoHelper {
+
   private Identifier identifier;
   private com.datastrato.gravitino.rel.Table gravitinoTable;
-  private TableCatalog sparkCatalog;
-  private Table lazySparkTable;
   private PropertiesConverter propertiesConverter;
   private SparkTransformConverter sparkTransformConverter;
 
-  public SparkBaseTable(
+  public GravitinoTableInfoHelper(
       Identifier identifier,
       com.datastrato.gravitino.rel.Table gravitinoTable,
-      TableCatalog sparkCatalog,
       PropertiesConverter propertiesConverter,
       SparkTransformConverter sparkTransformConverter) {
     this.identifier = identifier;
     this.gravitinoTable = gravitinoTable;
-    this.sparkCatalog = sparkCatalog;
     this.propertiesConverter = propertiesConverter;
     this.sparkTransformConverter = sparkTransformConverter;
   }
 
-  @Override
-  public String name() {
-    return getNormalizedIdentifier(identifier, gravitinoTable.name());
+  public String name(boolean isCaseSensitive) {
```

Comment:
move `isCaseSensitive` to  `GravitinoTableInfoHelper` construct parameters?

Reply:
done